libbeat/monitoring/inputmon/httphandler.go

```diff
@@ -81,19 +80,45 @@ func filteredSnapshot(r *monitoring.Registry, requestedType string) []map[string
 		}
 
 		// Require all entries to have an 'input' and 'id' to be accessed through this API.
-		if id, ok := m["id"].(string); !ok || id == "" {
+		id, ok := m["id"].(string)
+		if !ok || id == "" {
+			continue
+		}
+
+		if filterOutInput(m, requestedType) {
```

Comment:
I'd rather this helper was called something more descriptive like `validInputType`, and that it accepted the actual field value instead of the full dict, so it would be called like e.g. `if validInputType(m["input"], requestedType) ...`

Reply:
ok, I'll change that

Reply:
done: https://github.com/elastic/beats/pull/42618/commits/da345c0f59c295112598014dd572d3a5a69f6819